source/4.0/guides/sql-how-to/order-by-nested-association-attribute.html.md

```diff
@@ -0,0 +1,53 @@
+---
+chapter: SQL How To
+title: Order with attribute of an associated record
+---
+This article assumes:
+
+* You have a `users` table
+* You have a `teams` table 
+
+We have Users, each users is associated to a Team. Let's imagine we want to get all the users, 
+but we want them ordered by Team name, and if they're in the same team, we want them to be 
+ordered by creation date.
+
+``` ruby
+module Repositories
+  class Users
+    def get_ordered_by_team_and_creation_date
+      users
+        .join(:teams)
+        .order { |r| [r[:teams][:name], r[:users][:created_at]] }
```

Comment:
`order(teams[:name], teams[:created_at])` would be simpler

Reply:
I don't understand why I get that:
```
2.5.3 :007 > user_repo.users.left_join(:teams).order(teams[:name])
Traceback (most recent call last):
        2: from ./app.rb:5:in `<main>'
        1: from (irb):7
NameError (undefined local variable or method `teams' for main:Object)
```
Here is my Relations' code:
```
module Relations
  class Users < ROM::Relation[:sql]
    schema(:users) do
      attribute :id, ROM::Types::Integer
      attribute :name, ROM::Types::String
      attribute :created_at, ROM::Types::DateTime
      attribute :team_id, ROM::Types::Integer
      primary_key :id

      associations do
        belongs_to :team
      end
    end
  end

  class Teams < ROM::Relation[:sql]
    schema(:teams) do
      attribute :id, ROM::Types::Integer
      attribute :name, ROM::Types::String
      primary_key :id

      associations do
        has_many :users
      end
    end
  end
end
```
Is there anything I missed?
Plus, I want to order by the `name` of the team, and then by `created_at` attribute of my user

Reply:
You tried to access a relation in the context of the main object, so it crashed. What you probably meant to do was:

```ruby
user_repo.users.left_join(:teams).order(user_repo.teams[:name])
```

...but as I said, no need to use repos in this guide.

Reply:
Oh ok! I got it now! Thanks!